icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java

```diff
@@ -368,6 +368,24 @@ private void getAllowedHourFormats(ULocale uLocale) {
 
         String[] list = getAllowedHourFormatsLangCountry(language, country);
 
+        // We need to check if there is an hour cycle on locale
+        Character defaultCharFromLocale = null;
+        String hourCycle = uLocale.getKeywordValue("hours");
+        switch(hourCycle) {
```

Comment:
This line is triggering a NPE.  You should probably just check for `null` before entering your switch.  Maybe adding a `default` would also solve the problem.

Reply:
Oops. I though `switch` would handle this. I also added a case to test this path on new test.